src/app/modules/webhook/webhook.service.ts

```diff
@@ -217,3 +224,51 @@ const getFormattedResponse = (
     },
   }
 }
+
+export const pushData = (
+  webhookUrl: WebhookParams['webhookUrl'],
+  submissionWebhookView: WebhookView,
+): Promise<any> => {
+  const now = Date.now()
+  // Log and return, this should not happen.
+  if (!submissionWebhookView) {
+    logWebhookFailure(
+      new WebhookValidationError('submissionWebhookView was null'),
+      {
+        webhookUrl,
+        submissionWebhookView,
+        now,
+      },
+    )
+    return
+  }
+
+  const { submissionId, formId } = submissionWebhookView.data
+
+  const signature = formsgSdk.webhooks.generateSignature({
+    uri: webhookUrl,
+    submissionId,
+    formId,
+    epoch: now,
+  }) as string
```

Comment:
Why is there a need to cast this as string?

Reply:
`generateSignature` returns `string | void` but `webhookParams` takes in only `string`